### PR TITLE
[13.0][FIX] dms: /my/dms change to user and separate some urls in some functions

### PR DESCRIPTION
**Pull request by victoralmau**
PR in V12: https://github.com/OCA/dms/pull/29

Separate urls in some functions

- "my/dms/" changes to auth=user
- "my/dms/directory/<int:dms_directory_id>" optimized and simplified.

Please @pedrobaeza and @etobella  review it

@Tecnativa TT26319

### CODE REVIEW

File: dms/controllers/portal.py [hunk 1]
```diff
@@ -38,12 +38,69 @@ def _prepare_portal_layout_values(self):
...
+        items = (
+            request.env["dms.directory"]
+            .with_user(request.env.user.id)
+            .search(domain, order=sort_br)
+        )
+        request.session["my_dms_folder_history"] = items.ids
+        # values
+        values.update(
+            {
+                "dms_directories": items,
```
Comment (pedrobaeza) [inline dms/controllers/portal.py]: In v12 this is:

```suggestion
                "dms_directories": items.sudo(),
```

File: dms/controllers/portal.py [hunk 2]
```diff
@@ -38,12 +38,69 @@ def _prepare_portal_layout_values(self):
...
+                "page_name": "dms_directory",
+                "default_url": "/my/dms/directories",
```
Comment (pedrobaeza) [inline dms/controllers/portal.py]: In v12 this is:
```suggestion
                "default_url": "/my/dms",
```

File: dms/controllers/portal.py [hunk 3]
```diff
@@ -75,85 +131,64 @@ def portal_my_dms(
...
             search_domain = []
             if search_in == "name":
                 search_domain = OR([search_domain, [("name", "ilike", search)]])
             domain += search_domain
         # content according to pager and archive selected
-        items = (
+        dms_directory_items = (
             request.env["dms.directory"]
-            .with_user(request.env.user.id)
+            .sudo(request.env.user.id)
```
Comment (pedrobaeza) [inline dms/controllers/portal.py]:
```suggestion
            .with_user(request.env.user.id)
```

File: dms/controllers/portal.py [hunk 4]
```diff
@@ -75,85 +131,64 @@ def portal_my_dms(
...
-            )
+        request.session["my_dms_file_history"] = dms_file_items.ids
+        dms_parent_categories = dms_directory_sudo.sudo(
+            request.env.user.id
+        )._get_parent_categories(access_token)
+        # values
+        values = {
+            "dms_directories": dms_directory_items.sudo(),
+            "page_name": "dms_directory",
+            "default_url": "/my/dms/directories",
```
Comment (pedrobaeza) [inline dms/controllers/portal.py]:
```suggestion
            "default_url": "/my/dms",
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: dms/controllers/portal.py (hunk 3 above)
Comment (victoralmau) [reply dms/controllers/portal.py]: In V13 is needed use `with_user(id)` instead of `sudo(id)`.

File: dms/controllers/portal.py (hunk 4 above)
Comment (victoralmau) [reply dms/controllers/portal.py]: In V12 fix it isn't change.

File: dms/controllers/portal.py (hunk 3 above)
Comment (pedrobaeza) [reply dms/controllers/portal.py]: Exactly, that's why I'm requesting this change...

File: dms/controllers/portal.py (hunk 4 above)
Comment (pedrobaeza) [reply dms/controllers/portal.py]: Yeah, I know. It's not critical and I have seen it later, but better to put a valid URL in this version.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Merging as well:

/ocabot merge patch

**Pull request merged**